File: LeetCode/Easy/704 - Binary Search/704 - Binary Search.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var search = function(nums, target) {
    let lo = 0, hi = nums.length;
    while (lo <= hi) {
        let mid = Math.floor((lo + hi) / 2);
        
        if (nums[mid] == target) return mid;
        
        if (nums[mid] > target)
            hi = mid - 1;
        else 
            lo = mid + 1;
    }
    return -1;
};

File: LeetCode/Easy/704 - Binary Search/704 - Binary Search.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int lo = 0, hi = nums.size() - 1, mid;
        while (lo <= hi) {
            mid = lo + ((hi - lo) >> 1);
            if (nums[mid] == target) return mid;
            if (nums[mid] > target) hi = mid - 1;
            else lo = mid + 1;
        }
        return -1;
    }
};